node/core/dispute-coordinator/src/scraping/mod.rs

```diff
@@ -185,35 +169,24 @@ impl ChainScraper {
 
 	/// Prune finalized candidates.
 	///
-	/// Once a candidate lives in a relay chain block that's behind the finalized chain/got
-	/// finalized, we can treat it as low priority.
+	/// Once a candidate lives in a relay chain block that's behind `CANDIDATE_LIFETIME_AFTER_FINALIZATION`
+	/// blocks after finalization we can treat it as low priority.
 	pub fn process_finalized_block(&mut self, finalized_block_number: &BlockNumber) {
-		// First remove any stale backed and not included candidates
-		// Stale means any backed but not included candidates after `BACKED_CANDIDATE_MAX_LIFETIME`
-		// has passed since the (finalized) block containing the corresponding `CandidateBacked`
-		// event.
-		// `BACKED_CANDIDATE_MAX_LIFETIME - 1` because `finalized_block_number` counts to the backed
-		// candidate lifetime. Example:
-		// finalized_block_number = 4; BACKED_CANDIDATE_MAX_LIFETIME = 2;
+		// `CANDIDATE_LIFETIME_AFTER_FINALIZATION - 1` because `finalized_block_number`
+		// counts to the candidate lifetime.
+		// Example:
+		// finalized_block_number = 4; CANDIDATE_LIFETIME_AFTER_FINALIZATION = 2;
 		// key_to_clean = 4 - (2 - 1) = 3
 		// After `remove_all_at_block_height` at 3:
 		// 0, 1, 2 will be removed
 		// 3, 4 will be kept
-		// => We keep backed candidates in the last two finalized blocks
+		// => We keep candidates in the last two finalized blocks
 		finalized_block_number
-			.checked_sub(Self::BACKED_CANDIDATE_MAX_LIFETIME - 1)
-			.map(|key_to_clean| self.backed_candidates.remove_up_to_height(&key_to_clean));
-
-		// Then handle included candidates
-		let not_finalized =
-			self.included_candidates_by_block_number.split_off(finalized_block_number);
-		let finalized = std::mem::take(&mut self.included_candidates_by_block_number);
-		self.included_candidates_by_block_number = not_finalized;
-		// Clean up finalized
-		for finalized_candidate in finalized.into_values().flatten() {
-			self.included_candidates.remove(&finalized_candidate);
-			self.backed_candidates.remove(&finalized_candidate);
-		}
+			.checked_sub(Self::CANDIDATE_LIFETIME_AFTER_FINALIZATION - 1)
+			.map(|key_to_clean| {
```

Comment:
I'd personally prefer an `if let` or even an explicit `match` here, with a comment why we don't need to do any cleanup in the `None` case.